### PR TITLE
Remove paftl from the ntf parser

### PR DESCRIPTION
**Pull request by orange-vertex**
The current version does not really parse NFT files found on [edina digimap](https://digimap.edina.ac.uk/). Thus these changes along with the sample map reflect the User guide PDF found [here](http://webarchive.nationalarchives.gov.uk/20080901234220/http://www.ordnancesurvey.co.uk/oswebsite/products/landline/techinfo.html)

### CODE REVIEW

File: salalib/ntfp.cpp [hunk 1]
```diff
@@ -73,15 +76,15 @@ void NtfMap::fitBounds(const Line& li)
    }
 }
 
-void NtfMap::addGeom(int layer, NtfGeometry& geom)
+void NtfMap::addGeom(NtfLayer& layer, NtfGeometry& geom)
```
Comment (blsemo) [inline salalib/ntfp.cpp]: The only internal state this function changes with the new implementation is `m_line_count` - I am not sure that is appropriate if the layer itself gets passed in - the function is protected, so a derived class could pass in a completely unrelated layer and the internal state of the class would be all over the shop.
I think this function needs to either work on class data, or be a completely separate helper that is not a class member (and e.g. the count encountered here so call would be `m_line_count += addGeom(layer, geom);`)

File: salalib/ntfp.cpp [hunk 2]
```diff
@@ -100,16 +103,16 @@ void NtfMap::open(const std::vector<std::string>& fileset, Communicator *comm)
...
 /*
    m_bottom_left.a =  2147483647;   // 2^31 - 1
    m_bottom_left.b =  2147483647;
    m_top_right.a   = -2147483647;
    m_top_right.b   = -2147483647;
 */
    m_line_count = 0;
-   if (size()) {
-      clear();
+   if (!layers.empty()) {
```
Comment (blsemo) [inline salalib/ntfp.cpp]: The empty check is pointless, clear on an empty vector is no more expensive than the isEmpty check.

File: salalib/ntfp.h [hunk 3]
```diff
@@ -14,43 +15,46 @@
...
-#ifndef __NTFP_H__
-#define __NTFP_H__
+#include "genlib/p2dpoly.h"
 
 struct NtfPoint {
-   char m_chars;
+   int m_chars;
    int a;
    int b;
-   NtfPoint(char chars = 10)  // apparently 10 is NTF default
```
Comment (blsemo) [inline salalib/ntfp.h]: Have you double checked that this class does never get serialized into any file? Changing the variable size can play havoc with serialisation...

Comment (orange-vertex) [reply salalib/ntfp.h]: Yes, this class is only used as a local temporary variable to load NTF files in MetaGraph::loadLineData

**Pull request merged**